Makefile-switchroot.am

```diff
@@ -28,9 +27,32 @@ libswitchroot_mountutil_la_SOURCES = \
 	src/switchroot/ostree-mount-util.h \
 	$(NULL)
 
-ostree_prepare_root_SOURCES = src/switchroot/ostree-prepare-root.c
-ostree_prepare_root_LDADD = libswitchroot-mountutil.la
+ostree_prepare_root_SOURCES = \
+    src/switchroot/ostree-mount-util.c \
+    src/switchroot/ostree-mount-util.h \
+    src/switchroot/ostree-prepare-root.c \
+    $(NULL)
+
+if BUILDOPT_USE_STATIC_COMPILER
+# ostree-prepare-root can be used as init in a system without a populated /lib.
+# To support this use case we need to link statically as we will be unable to
+# locate libc.so at run time if it's not installed in /lib.
+#
+# We support building ostree-prepare-root with a different compiler to the rest
+# of ostree so we can use musl rather than glibc.  This reduces the size of the
+# executable significantly: from ~700K -> ~30K.  We have to use _SCRIPTS here
+# to get autotools to install this as an executable but without generating rules
+# to make it itself which we have specified manually.  See
+# https://lists.gnu.org/archive/html/help-gnu-utils/2007-01/msg00007.html
+ostree_boot_SCRIPTS = ostree-prepare-root
+
+ostree-prepare-root : $(ostree_prepare_root_SOURCES)
+	$(STATIC_COMPILER) -o $@ -static $(ostree_prepare_root_SOURCES) $(AM_CPPFLAGS) $(AM_CFLAGS)
+else
+ostree_boot_PROGRAMS += ostree-prepare-root
+
 ostree_prepare_root_CFLAGS = $(AM_CFLAGS) -Isrc/switchroot
+endif
 
 ostree_remount_SOURCES = src/switchroot/ostree-remount.c
 ostree_remount_LDADD = libswitchroot-mountutil.la
```

Comment:
Probably.  One thing I meant to comment on - it looks like we can switch to `err` rather than `perrorv + exit()`?  That would clean up the code a lot, then we could likely move `path_is_on_readonly_fs` to a static inline or so.